source/OptimaJet.DWKit.StarterApplication/Services/ProjectService.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal;
+using JsonApiDotNetCore.Services;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class ProjectService : EntityResourceService<Project>
+    {
+        public IOrganizationContext OrganizationContext { get; private set; }
+        public IJsonApiContext JsonApiContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public UserRepository UserRepository { get; }
+        public GroupRepository GroupRepository { get; }
+        public ProjectRepository ProjectRepository { get; }
+        public IEntityRepository<Organization> OrganizationRepository { get; set; }
+
+        public ProjectService(
+            IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
+            UserRepository userRepository,
+            IEntityRepository<Project> projectRepository,
+            GroupRepository groupRepository,
+            IEntityRepository<Organization> organizationRepository,
+            ILoggerFactory loggerFactory) : base(jsonApiContext, projectRepository, loggerFactory)
+        {
+            OrganizationContext = organizationContext;
+            JsonApiContext = jsonApiContext;
+            CurrentUserContext = currentUserContext;
+            UserRepository = userRepository;
+            GroupRepository = groupRepository;
+            OrganizationRepository = organizationRepository;
+            ProjectRepository = (ProjectRepository)projectRepository;
+        }
+        public override async Task<IEnumerable<Project>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<Project>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
```

Comment:
I've seen this in a few places. I wonder if there should be a method on the OrganizationContext.

Reply:
With the GetAsync method extraction, this only appears in one place now